resources/migrations/000_migrations.yaml

```diff
@@ -8648,6 +8657,846 @@ databaseChangeLog:
                   constraints:
                     nullable: false
                   defaultValue: false
+  - changeSet:
+      id: 321
+      author: jeff303
+      comment: Added 0.42.0 - modify type of activity.details from text to ${text.type}
+        on mysql,mariadb
+      changes:
+        - modifyDataType:
+            tableName: activity
+            columnName: details
+            newDataType: ${text.type}
+      preConditions:
+        - onFail: MARK_RAN
+        - dbms:
+            type: mysql,mariadb
```

Comment:
I'm not 100% sure that you can list multiple types here, or if you need to use `or`... have you manually confirmed that this works for both MySQL and MariaDB? Maybe adding a test for MySQL/MariaDB that just confirms the application DB columns are the right types or something could be a good idea

Reply:
The same thing is used in a couple other places in the file already, [ex](https://github.com/metabase/metabase/blame/a459f19ac8805614acb4709ae16e3aacafb2776a/resources/migrations/000_migrations.yaml#L5342-L5345).  But I can add a test (looks like `test-migrations` macro should work for this).

Reply:
You probably don't even need `test-migrations`, just write a test that checks the type of one of the app DB columns if we're running MySQL or MariaDB. You can use JDBC `DatabaseMetaData` to do that

Reply:
I just wrote a comprehensive test for all app DBs.